src/ConnectedMode.UnitTests/Binding/BindingProcessImplTests.cs

```diff
@@ -156,190 +111,20 @@ private static ServerExclusions CreateSettings()
         }
 
         [TestMethod]
-        public async Task DownloadQualityProfile_SavesConfiguration()
-        {
-            var configPersister = new DummyConfigPersister();
-
-            // Arrange
-            const string QualityProfileName = "SQQualityProfileName";
-            const string ProjectName = "SQProjectName";
-
-            var progressAdapter = Mock.Of<IProgress<FixedStepsProgress>>();
-
-            var bindingConfig = new Mock<IBindingConfig>().Object;
-
-            var language = Language.VBNET;
-            var sonarQubeService = new Mock<ISonarQubeService>();
-            SonarQubeQualityProfile profile = ConfigureQualityProfile(sonarQubeService, language, QualityProfileName);
-
-            var configProviderMock = new Mock<IBindingConfigProvider>();
-            configProviderMock.Setup(x => x.GetConfigurationAsync(profile, language, It.IsAny<BindingConfiguration>(), CancellationToken.None))
-                .ReturnsAsync(bindingConfig);
-
-            var bindingArgs = CreateBindCommandArgs("key", ProjectName, new ConnectionInformation(new Uri("http://connected")));
-            var testSubject = CreateTestSubject(bindingArgs,
-                configProviderMock.Object,
-                sonarQubeService: sonarQubeService.Object,
-                configurationPersister: configPersister,
-                languagesToBind: new []{ language });
-
-            // Act
-            await testSubject.DownloadQualityProfileAsync(progressAdapter, CancellationToken.None);
-
-            // Assert
-            configPersister.SavedProject.Should().NotBeNull();
-
-            var savedProject = configPersister.SavedProject;
-            savedProject.ServerUri.Should().Be(bindingArgs.Connection.ServerUri);
-            savedProject.Profiles.Should().HaveCount(1);
-            savedProject.Profiles[Language.VBNET].ProfileKey.Should().Be(profile.Key);
-            savedProject.Profiles[Language.VBNET].ProfileTimestamp.Should().Be(profile.TimeStamp);
-        }
-
-        [TestMethod]
-        public async Task DownloadQualityProfile_WhenQualityProfileIsNotAvailable_OtherLanguagesDownloadedSucceessfully()
```

Comment:
I copied this test, then modified it be extracting all of the progress-checking code to a separate test.